string Action
int8 ActionIdx
